src/NHibernate/Transaction/AdoTransaction.cs

```diff
@@ -186,6 +186,13 @@ public void Commit()
 
 				log.Debug("Start Commit");
 
+				foreach (var dependentSession in session.ConnectionManager.DependentSessions)
```

Comment:
Reverted? No, that is precisely the main point of this PR.
Reversed order? I do not get why.